### DISCUSSION TITLE
Add an option to make unlit VolumeFog invisible

### DISCUSSION THREAD

**Discussion by art-oria** [godotengine/godot-proposals#4882] [3D]
Making volumetric fog not appear when not hit by a light (example from Unity where this is the case) allows for effects such as fake Godrays that do not darken the scene
![Screenshot_3](https://user-images.githubusercontent.com/37030911/179307261-e281574c-565e-4003-974f-c40e84a6e5ae.png)

In Godot when fog is not hit by any light you get black fog that blocks the view
![Screenshot_1](https://user-images.githubusercontent.com/37030911/179307440-d9ca7990-bb91-4e1b-9077-53a5fc5efe7a.png)

An option to have fog disappear when not hit by a light of a specific layer would also allow for fogs of more specific shape maybe
![man-dog-searching-flashlight-outdoor-exploring-field-night-lightening-torch-to-sky-169227548](https://user-images.githubusercontent.com/37030911/179308378-5c205d67-17db-4c60-b068-6e2e4c805793.jpg)

**Comment by Calinou** [comment]
Related to https://github.com/godotengine/godot-proposals/issues/1374.

As I described in the linked proposal, I started working on a property that allows you to use volumetric fog in a purely additive way, effectively turning it into a volumetric *lighting* solution. However, this needs to have per-light control on volumetric fog energy for this to be usable.

> An option to have fog disappear when not hit by a light of a specific layer would also allow for fogs of more specific shape maybe

The particular example you've shown can be achieved using a FogVolume node set to a Cone shape and with an emissive color matching the light's color.
